python/pyspark/sql/pandas/serializers.py

```diff
@@ -215,7 +215,10 @@ def _create_batch(self, series):
         series = ((s, None) if not isinstance(s, (list, tuple)) else s for s in series)
 
         def create_array(s, t):
-            mask = s.isnull()
+            if hasattr(s.values, "__arrow_array__"):
```

Comment:
this actually reminds me of https://github.com/apache/spark/pull/28743 how is it related to each other?

Reply:
Looks really similar. I've added my tests to that branch and they passed. (However, lot of other tests failed but probably for other reasons).

Reply:
I think it's preferred to use `s.array` instead of `s.values`, would you mind giving that a try?